packages/x-utils/src/typeguards.ts

```diff
@@ -25,3 +27,22 @@ export function isFunction(value: any): value is AnyFunction {
 export function isObject(value: any): value is Record<string, unknown> {
   return Boolean(value) && typeof value === 'object' && !Array.isArray(value);
 }
+
+/**
+ * Determines whether the passed path is a valid property path of the passed object.
+ *
+ * @param obj - The object to check the path against.
+ * @param path - The path to be checked.
+ *
+ * @returns `true` if `path` is a valid path of `obj`; otherwise, `false`.
+ *
+ * @public
+ */
+export function isPath<SomeObject, Path extends ExtractPath<SomeObject>>(
+  obj: SomeObject,
+  path: string
+): path is Path {
+  return getSafePropertyChain<ExtractType<SomeObject, Path>>(obj, path) !== undefined
```

Comment:
this can be simplify by returning `getSafePropertyChain<ExtractType<SomeObject, Path>>(obj, path) !== undefined`